### PR TITLE
feat: Error handling + test utils

### PR DESCRIPTION
**Pull request by dgca**
- Adds service method util to handle uncaught errors. [link](https://github.com/iron-fish/wallet-
server/pull/16/files#diff-9153efea0f4f7d4986b19ecdc74b7be162f54cc871d64bf8ad3c59fec87ee308R1)
- Fixes running integration tests by closing leveldown db connection on teardown

### CODE REVIEW

File: src/cache/index.ts [hunk 1]
```diff
@@ -3,9 +3,9 @@ import leveldown from "leveldown";
 import levelup, { LevelUp } from "levelup";
 import path from "path";
 
-import { ifClient } from "../utils/ironfish";
-import { lightBlock } from "../utils/light_block";
```
Comment (dgca) [inline src/cache/index.ts:7]: Figured I'd camelCase this for consistency.

File: src/services/LightStreamer.ts [hunk 2]
```diff
@@ -8,63 +8,59 @@ import {
...
-    call,
-    callback,
-  ) => {
+  public getBlock = handle<BlockID, LightBlock>(async (call, callback) => {
     if (!call.request.hash && !call.request.sequence) {
-      callback(new Error("Must provide either hash or sequence"), null);
+      throw new ServiceError(
+        status.INVALID_ARGUMENT,
+        "Either hash or sequence must be provided",
+      );
```
Comment (dgca) [inline src/services/LightStreamer.ts:32]: The `handle` function lets you throw errors instead having to `callback(err, null)`.

File: src/utils/lightBlock.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import { GetBlockResponse, FollowChainStreamResponse } from "@ironfish/sdk";
```
Comment (dgca) [inline src/utils/lightBlock.ts:1]: Only renamed this file.

File: .vscode/settings.json [hunk 4]
```diff
@@ -0,0 +1,3 @@
+{
+  "typescript.tsdk": "node_modules/typescript/lib"
```
Comment (jowparks) [inline .vscode/settings.json:2]: Out of curiosity what does this do?

File: src/utils/lightBlock.ts (hunk 3 above)
Comment (jowparks) [reply src/utils/lightBlock.ts:1]: nit: if you use `git mv` this diff shouldn't happen. Also I think if you move via VSCode UI I would think this diff shouldn't happen.

**Review by jowparks — APPROVED** [review]

File: .vscode/settings.json (hunk 4 above)
Comment (dgca) [reply .vscode/settings.json:2]: VSCode will usually just use whatever TS version it's currently at for type checking. This forces it to use the version installed in the project via package.json.

**Pull request merged**